### DISCUSSION TITLE
Auth UserFields remapping does not work.

### DISCUSSION THREAD

**Discussion by cyborg-x1** [lowdefy/lowdefy#1564] [Questions and Help]
#### Describe the bug

Using KeycloakProvider, Login works as expected.

But I can't get auth.userFields to work. I tried remapping it in Keycloak directly as roles field in the token.
But it does not show up in the _user operator. 

 I remapped the roles to profile.roles in keycloak and it shows up also with the operator in profile.roles. I tried to remap it according to the latest documentation preview of the latest pullrequest (Example -> roles: profile.myroles) 

Result: roles are not working and roles field is not visible in user operator.

I also tried to remap any existing user field to another like for example email to my_email, which also does not work.

If for example I remap the roles in keycloak to a field like email - its overwritten and the roles show up there.

I can not get any field from the provider to show up which does not exist here:
https://github.com/nextauthjs/next-auth/blob/main/packages/next-auth/src/providers/keycloak.ts#L3-L25

#### To Reproduce

Try to remap a existing field to a not existing field with userFields.

#### Lowdefy Version

Version 4.0.0-rc.5 

#### Additional Info
I tried to find out from the code how it works and maybe found an issue:

In those two locations:
https://github.com/lowdefy/lowdefy/blob/main/packages/api/src/routes/auth/callbacks/addUserFieldsToSession.js#L19
https://github.com/lowdefy/lowdefy/blob/main/packages/api/src/routes/auth/callbacks/addUserFieldsToSession.js#L24

you use only the key **but not the value (the field from which the value should come from)** of the auth.userFields. 
Is that correct?

While here you do use both: https://github.com/lowdefy/lowdefy/blob/main/packages/api/src/routes/auth/callbacks/addUserFieldsToToken.js#L24-L31

**Comment by SamTolmay** [comment]
Hi @cyborg-x1 

Sorry, I missed this issue so only replying now. 

The code the is correct, first we map the value from next-auth to the token, and then the value in the token to the session. It has been working for other providers, but we haven't tested it with Keycloak

Have you tried mapping the entire account and profile objects like this?
```yaml
  userFields:
    account: account
    profile: profile
```
